src/util/string.cpp

```diff
@@ -8,6 +8,25 @@
 #include <string>
 
 namespace util {
+
+// Compile-time sanity checks
+static_assert([] {
```

Comment:
This would pass if we forgot to add an exception, we need to test the negatives as well. This is the reason for my NACK